mgmt/rest/rest_func_test.go

```diff
@@ -582,6 +583,12 @@ func TestPluginRestCalls(t *testing.T) {
 				cfg := getDefaultMockConfig()
 				err := cfgfile.Read("../../examples/configs/snap-config-sample.json", &cfg, MOCK_CONSTRAINTS)
 				So(err, ShouldBeNil)
+				if len(SNAP_AUTODISCOVER_PATH) == 0 {
+					if len(SNAP_PATH) != 0 {
+						SNAP_AUTODISCOVER_PATH = fmt.Sprintf("%s/plugin", SNAP_PATH)
+					}
+				}
+				cfg.Control.AutoDiscoverPath = SNAP_AUTODISCOVER_PATH
```

Comment:
This causes failures if the SNAP_AUTODISCOVER_PATH is set to something where the plugins fail to load (since failures on when autoloading plugins are fatal). 

What is the reasoning behind adding this/the SNAP_AUTODISCOVER_PATH env variable to this test?